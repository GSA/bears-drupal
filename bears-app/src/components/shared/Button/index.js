import './index.scss'

/**
 * a functional component that renders a reactive button
 * @component
 * @param {object} children - inherited children
 * @param {string} className - inherited class(es)
 * @param {event} onClick - an inherited function, triggered on a click event
 * @return {html} returns a semantic html button element
 */
function Button({ children, className, onClick }) {
  /**
   * a function to construct class strings based on composition.
   * @function
   * @param {string} className - The inherited class
   * @return {string} returns a concat string
   */
  const handleClassName =
    className !== undefined ? `usa-button ${className}` : 'usa-button'

  return (
    <button onClick={onClick} className={handleClassName}>
      {children}
    </button>
  )
}

export default Button
