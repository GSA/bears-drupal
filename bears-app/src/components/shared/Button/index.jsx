import PropTypes from 'prop-types'

import './index.scss'

/**
 * a functional component that renders a reactive button
 * @component
 * @param {React.ReactNode} children - inherited children
 * @param {string} className - inherited class(es)
 * @param {function} onClick - an inherited function, triggered on a click event
 * @return {html} returns a semantic html button element
 */
function Button({ children, className, onClick }) {
  /**
   * a function to construct class strings based on composition.
   * @function
   * @param {string} className - The inherited class
   * @return {string} returns a concat string
   */
  const handleClassName =
    className !== undefined ? `usa-button ${className}` : 'usa-button'

  return (
    <button onClick={onClick} className={handleClassName}>
      {children}
    </button>
  )
}

Button.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  onClick: PropTypes.func,
}

export default Button
