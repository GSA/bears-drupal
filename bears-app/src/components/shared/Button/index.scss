// this is an example of using sass within our styles
.usa-button {
&.lighter {
    background-color: lightblue;





       }
}
