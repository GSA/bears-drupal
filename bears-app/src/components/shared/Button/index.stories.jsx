import Button from './index.jsx'

export default {
  component: Button,
}

export const Primary = {
  render: () => <Button>Button</Button>,
}

export const Secondary = {
  render: () => <Button className="border-2px">Button</Button>,
}
