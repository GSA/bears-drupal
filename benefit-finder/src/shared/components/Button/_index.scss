@use '../../styles/colors/_index.scss' as color;

#benefit-finder {
  .usa-button {
    background-color: color.$pop-blue !important;
    color: color.$white !important;
  }
  /* stylelint-disable selector-class-pattern */
  .usa-button--unstyled {
    background-color: transparent !important;
    color: color.$pop-blue !important;
  }

  .usa-button.usa-button--unstyled:hover {
    background-color: transparent !important;
    color: color.$pop-blue !important;
  }

  .usa-button:hover {
    background-color: color.$marine !important;
    color: color.$white !important;
  }

  .usa-button--outline {
    background-color: color.$white !important;
    box-shadow: inset 0 0 0 2px color.$pop-blue !important;
    color: color.$pop-blue !important;
  }

  .usa-button--outline:hover {
    background-color: color.$white !important;
    box-shadow: inset 0 0 0 2px color.$marine !important;
    color: color.$marine !important;
  }
  /* stylelint-enable selector-class-pattern */
}
