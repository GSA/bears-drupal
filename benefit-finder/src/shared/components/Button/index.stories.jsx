import Button from './index.jsx'

export default {
  component: Button,
  tags: ['autodocs'],
  args: {
    children: 'Button Text',
    disabled: false,
    type: 'button',
  },
}

export const Default = {}

export const Primary = {
  args: {
    children: 'Button',
  },
}

export const Secondary = {
  args: {
    ...Primary.args,
    secondary: true,
  },
}

export const Unstyled = {
  args: {
    ...Primary.args,
    unstyled: true,
  },
}
