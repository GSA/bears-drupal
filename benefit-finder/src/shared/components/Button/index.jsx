import { useEffect, useState } from 'react'
import { useHandleClassName } from '../../hooks/useHandleClassName'
import PropTypes from 'prop-types'
import './_index.scss'

/**
 * a functional component that renders a reactive button
 * @component
 * @param {React.ReactNode} children - inherited children
 * @param {string} className - inherited class(es)
 * @param {boolean} secondary - alternative display styles
 * @param {boolean} disabled - disabled state
 * @param {boolean} unstyled - appear as a link
 * @param {function} onClick - an inherited function, triggered on a click event
 * @return {html} returns a semantic html button element
 */

function Button({
  children,
  className,
  onClick,
  secondary,
  disabled,
  unstyled,
  type,
}) {
  const [defaultClasses, setDefaultClasses] = useState(null)
  const style =
    secondary === true ? 'secondary' : unstyled === true ? 'unstyled' : null

  useEffect(() => {
    switch (style) {
      case 'secondary':
        setDefaultClasses(['usa-button', 'usa-button--outline'])
        break
      case 'unstyled':
        setDefaultClasses(['usa-button', 'usa-button--unstyled'])
        break
      default:
        setDefaultClasses(['usa-button'])
    }
  }, [style, secondary, unstyled])

  return (
    <button
      onClick={disabled ? null : onClick}
      type={type || 'button'}
      disabled={disabled}
      aria-disabled={disabled}
      className={useHandleClassName({
        className,
        defaultClasses,
      })}
    >
      {children}
    </button>
  )
}

Button.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  secondary: PropTypes.bool,
  unstyled: PropTypes.bool,
  onClick: PropTypes.func,
  type: PropTypes.oneOf(['button', 'reset', 'download']),
}

export default Button
