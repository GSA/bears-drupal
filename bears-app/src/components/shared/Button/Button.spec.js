import TestRenderer from 'react-test-renderer'
import Button from './index.jsx'

describe('Select', () => {
  it('renders a match to the previous snapshot', () => {
    const component = TestRenderer.create(<Button />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
