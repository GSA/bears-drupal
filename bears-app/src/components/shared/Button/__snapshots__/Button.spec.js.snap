// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders a match to the previous snapshot 1`] = `
<button
  className="usa-button"
/>
`;
